pkg/envoy/registry/registry.go

```diff
@@ -10,43 +10,66 @@ func NewProxyRegistry(mapper ProxyServiceMapper, msgBroker *messaging.Broker) *P
 	return &ProxyRegistry{
 		ProxyServiceMapper: mapper,
 		msgBroker:          msgBroker,
+		connectedProxies:   make(map[string]*envoy.Proxy),
 	}
 }
 
 // RegisterProxy registers a newly connected proxy.
 func (pr *ProxyRegistry) RegisterProxy(proxy *envoy.Proxy) {
-	pr.connectedProxies.Store(proxy.UUID.String(), proxy)
+	uuid := proxy.UUID.String()
+	if pr.GetConnectedProxy(uuid) != nil {
```

Comment:
The way in which a proxy can disconnect and reconnect in rapid succession would actually allow a proxy to still be "registered", prior to it being deregistered.

This is a pre-existing bug, and isn't quite in the critical path, so likely not a big concern, although we should make a note of it and consider addressing it later.

One solution would require a nonce, or a version number to check when unregistering, and only unregistering if those match.

Reply:
Let's create a new issue for this bug. Nonce might not be a good solution. Sequential version number should maintain the request sequence. 

E.g. a proxy sends unregister and register requests. Due to network delay, register request arrives earlier. Without sequence number, it will eventually be deregistered.

Reply:
Can you create the issue and add a TODO here referencing the issue number?

Reply:
New issue opened to track the bug: https://github.com/openservicemesh/osm/issues/4950

Reply:
Awesome! Did you add the TODO(#4950) comment?

Reply:
Just added. Let's wait for the CI pass.